app/src/main/java/com/github/se/eventradar/model/repository/message/FirebaseMessageRepository.kt

```diff
@@ -46,25 +79,37 @@ class FirebaseMessageRepository : IMessageRepository {
   ): Resource<Unit> {
     return try {
       val newMessage =
-          messageRef.document(messageHistory.id).collection("messages").add(message.toMap()).await()
-      messageRef.document(messageHistory.id).update("latest_message_id", newMessage.id).await()
+          messageRef
+              .document(messageHistory.id)
+              .collection("messages_list")
+              .add(message.toMap())
+              .await()
+
+      val updatedValues =
+          mapOf(
+              "latest_message_id" to newMessage.id,
+              "from_user_read" to (message.sender == messageHistory.user1),
+              "to_user_read" to (message.sender == messageHistory.user2),
+          )
+
+      messageRef.document(messageHistory.id).update(updatedValues).await()
       Resource.Success(Unit)
     } catch (e: Exception) {
       Resource.Failure(e)
     }
   }
 
-  override suspend fun updateMessageToReadState(
-      message: Message,
+  override suspend fun updateReadStateForUser(
+      userid: String,
       messageHistory: MessageHistory
   ): Resource<Unit> {
+    val updatedValue =
+        mapOf(
+            if (userid == messageHistory.user1) "from_user_read" to true
```

Comment:
This is not used yet because it is meant to be used in the private chat. It could be useful when one person sends a message and the other person goes to read it. Then this method can be called to only update their read status (as the original sender's status is already correct). If you think there is a better way to update the read state please let me know.

Reply:
Sounds good, asking as I was doing the Private Chat function, and now @Nath-Assa is doing it in Sprint 8